test/docker-e2e/e2e_upgrade_test.go

```diff
@@ -0,0 +1,294 @@
+func (s *CelestiaTestSuite) TestCelestiaAppSignalDrivenUpgrade() {
+	if testing.Short() {
+		s.T().Skip("skipping celestia-app major upgrade test in short mode")
+	}
+
+	tag, err := dockerchain.GetCelestiaTagStrict()
+	s.Require().NoError(err)
+
+	tt := []struct {
+		Name             string
+		ImageTag         string
+		TargetAppVersion uint64
+	}{
+		{
+			Name:             "v2 to v3",
+			ImageTag:         tag,
+			TargetAppVersion: 3,
+		},
+		{
+			Name:             "v3 to v4",
+			ImageTag:         tag,
+			TargetAppVersion: 4,
+		},
+	}
+
+	for _, tc := range tt {
+		s.Run(tc.Name, func() {
+			s.runSignalDrivenUpgradeTest(tc.ImageTag, tc.TargetAppVersion)
+		})
+	}
+}
+
+// runBinarySwapUpgradeTest tests a binary swap (minor version) upgrade for celestia-app.
+// It initializes a chain with the given base image tag, verifies bank send functionality,
+// upgrades the chain to the target image tag, restarts it, and then verifies both the
+// new binary version and continued bank send functionality.
+func (s *CelestiaTestSuite) runBinarySwapUpgradeTest(BaseImageTag, TargetImageTag string) {
+	var (
+		ctx = context.Background()
+		cfg = dockerchain.DefaultConfig(s.client, s.network).WithTag(BaseImageTag)
+	)
+
+	chain, err := dockerchain.NewCelestiaChainBuilder(s.T(), cfg).Build(ctx)
+	s.Require().NoError(err)
+
+	s.T().Cleanup(func() {
+		if err := chain.Stop(ctx); err != nil {
+			s.T().Logf("Error stopping chain: %v", err)
+		}
+	})
+
+	err = chain.Start(ctx)
+	s.Require().NoError(err)
+
+	// Sanity check: Test bank send before upgrade
+	s.T().Log("Testing bank send functionality before upgrade")
+	testBankSend(s.T(), chain, cfg)
+
+	err = chain.Stop(ctx)
+	s.Require().NoError(err)
+
+	chain.UpgradeVersion(ctx, TargetImageTag)
+
+	err = chain.Start(ctx)
+	s.Require().NoError(err)
+
+	// Sanity check: Test bank send after upgrade
+	s.T().Log("Testing bank send functionality after upgrade")
+	testBankSend(s.T(), chain, cfg)
+
+	// Verify the binary version after upgrade
+	validatorNode := chain.GetNodes()[0]
+
+	rpcClient, err := validatorNode.GetRPCClient()
+	s.Require().NoError(err, "failed to get RPC client for version check")
+
+	abciInfo, err := rpcClient.ABCIInfo(ctx)
+	s.Require().NoError(err, "failed to fetch ABCI info")
+	s.Require().Contains(abciInfo.Response.GetVersion(), strings.TrimPrefix(TargetImageTag, "v"), "version mismatch")
+}
+
+// runSignalDrivenUpgradeTest performs an end-to-end test of a major (signaled) upgrade for celestia-app.
+// It starts a chain at the given image tag with app version set to one less than the target,
+// signals all validators for the upgrade, ensures the upgrade is scheduled and executed,
+// and verifies the chain is running the new binary and app version after the upgrade.
+// The test also checks bank send functionality and that the voting power threshold is met before proceeding.
+func (s *CelestiaTestSuite) runSignalDrivenUpgradeTest(ImageTag string, TargetAppVersion uint64) {
```

Comment:
not blocking on naming, but what other types of upgrades are there beyond signaled? we don't support governance upgrades